internal/drivers/nomad/driver.go

```diff
@@ -495,7 +495,7 @@ func (p *config) Destroy(ctx context.Context, instances []*types.Instance) (err
 			WithField("job_id", jobID).WithField("resource_job_id", resourceJobID)
 
 		logr.Debugln("scheduler: freeing up resources ... ")
-		err = p.deregisterJob(logr, resourceJobID, true)
+		err = p.deregisterJob(logr, resourceJobID, false)
```

Comment:
can we make purge false in case provisioning failed ?

Reply:
it's already false everywhere